lib/sidekiq/client.rb

```diff
@@ -90,16 +90,16 @@ def push(item)
     # Returns an array of the of pushed jobs' jids.  The number of jobs pushed can be less
     # than the number given if the middleware stopped processing for one or more jobs.
     def push_bulk(items)
-      arg = items["args"].first
-      return [] unless arg # no jobs to push
-      raise ArgumentError, "Bulk arguments must be an Array of Arrays: [[1], [2]]" unless arg.is_a?(Array)
+      args = items["args"]
+      raise ArgumentError, "Bulk arguments must be an Array of Arrays: [[1], [2]]" unless (args.is_a?(Array) && args.all?(Array))
+      return [] if args.empty? # no jobs to push
```

Comment:
adding `.empty?` because `return [] unless args` continues execution even if the array is empty.
And this is now added after checking that arg is indeed an Array, to make sure that checks like `nil.empty?` never happen.